src/transformers/models/stablelm/modeling_stablelm.py

```diff
@@ -203,6 +203,21 @@ def forward(self, x):
         return self.down_proj(self.act_fn(self.gate_proj(x)) * self.up_proj(x))
 
 
+class StableLmLayerNormPerHead(nn.Module):
```

Comment:
this seems pretty similar to the `Persimmon` or `Phi` layer norm no? 

https://github.com/huggingface/transformers/blob/bbd26b001ad1f86a94129769cc61ebcaac416a3e/src/transformers/models/phi/modeling_phi.py#L273-L279

Reply:
No. This implementation of QK normalization is per-head, i.e., each head has its own independent learnable gamma (bias unused), whereas the phi/persimmon implementations are per-layer, sharing affine parameters across all heads.

Reply:
Thanks for explaining! Let's add this to the doc of this class! 😉 
Also does that mean the weights are shared for all layers? Or it's per head per layer?